docs/enhancements/easy-discovery-password.md

```diff
@@ -0,0 +1,174 @@
+
+In addition, we should improve the API and documnetation to make it possible
+for users to set an ignition `core` user password without having to deal with
+complicated ignition overrides.
+
+## Motivation
+
+The discovery ISO generated by the Assisted Installer comes with a `core` user
+that has its password login disabled (although SSH access using the provided
+public key is still supported).
+
+However, it can sometimes be useful (e.g. when the network is misbehaving, SSH
+isn't working, and you want to troubleshoot why), to login to the machine
+directly (with physical or through server remote management solutions), using a
+password. This can be done using the console login prompt that is shown after
+RHCOS is done booting.
+
+To allow that, the user must have a password set. In order to set a password,
+it's possible to use the Assisted Installer ignition override API which
+provides support for editing the discovery ISO's ignition file with custom
+parameters.
+
+This method is complicated and should be made easier.
+
+### Goals
+
+- Add new UI to set the `core` user password for the discovery ISO 
+- Add new API to set the ignition password hash for the discovery ISO `core`
+user password, with documentation explaining how it can be done
+- Do all of that in a secure way. At no point should the service know the
+user's password. But it's okay if the service knows the hash, as that doesn't
+reveal the password
+
+### Non-Goals
+
+- Allow the core user password to be set even after the host has been
+discovered. We could technically create a new agent step that would set the
+discovery password hash to whatever the user wants on a live system, but that
+would be useless in 99% of cases since if the networking is working well enough
+for the agent to connect to the service, it's probably working well enough for
+the user to SSH into the server using their public key anyway - no password
+needed.
+
+## Proposal
+
+### New service APIs
+
+#### REST API
+
+- Add a new optional `discovery_core_user_password_hash` parameter to the
+`infra-env-create-params` API structure. Ensure this parameter has
+`description:` documentation explaining how the hash can be calculated and what
+its format should look like, preferably with an example. The service will use
+this new parameter to set the `core` user discovery ISO password hash. Since
+this is just a salted hash, the service has no practical way to figure out what
+the password actually is.
+
+#### Kubernetes API
+
+- Extend the InfraEnv CR to support a similar (see REST API section above)
+`discovery_core_user_password_hash` parameter.
+
+### UI changes
+
+- Extend the discovery ISO generation UI modal to let users input a password.
+If and when the user provides a password in that field, UI will calculate the
+hash for the user-provided password and will set the `discovery_core_user_password_hash`
+parameter accordingly.
```

Comment:
If the user exits that modal and returns, the UI can't show the password because the backend doesn't return it (not a bad thing, just a fact).  Suggest showing ********* if core_user_password_hash is not empty string.

Reply:
@nirfarkas should review this

Reply:
>the UI can't show the password because the backend doesn't return it

The UI can technically store the password in browser local storage, but I agree just forgetting the password is probably best

>Suggest showing ********* if core_user_password_hash is not empty string.

A bit problematic because we don't even know the length of the password

Reply:
It doesn't need to have the right amount of *, actually having it always show the same amount is more secure.  Anyway, @nirfarkas can figure out the best way to indicate that a password was set and to allow updating it.